### ISSUE TITLE
AVFoundation.AVCaptureDeviceInput Disposed

### ISSUE THREAD

**Issue by Martinchr82** [Redth/ZXing.Net.Maui#68]
When coming to the QR page, I allow access to use the Camera, which is working fine. But if I then go to Settings on my iPhone X and selects the app, and then toggle the camera access to off, and go into the app again. Then it crashes with the following StackTrace 

Cannot access a disposed object. Object name: 'AVFoundation.AVCaptureDeviceInput'

ObjCRuntime.ThrowHelper.ThrowObjectDisposedException(Object o)
ObjCRuntime.NativeObjectExtensions.GetNonNullHandle(INativeObject self, String argumentName)
AVFoundation.AVCaptureSession.AddInput(AVCaptureInput input)
ZXing.Net.Maui.CameraManager.UpdateCamera()
ZXing.Net.Maui.CameraManager.UpdateCameraLocation(CameraLocation cameraLocation)
ZXing.Net.Maui.CameraBarcodeReaderViewHandler.<>c.<.cctor>b__18_1(CameraBarcodeReaderViewHandler handler, ICameraBarcodeReaderView virtualView)
Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[ZXing.Net.Maui.ICameraBarcodeReaderView, ZXing.Net.MAUI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[ZXing.Net.Maui.CameraBarcodeReaderViewHandler, ZXing.Net.MAUI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
Microsoft.Maui.Handlers.ViewHandler`2[[ZXing.Net.Maui.ICameraBarcodeReaderView, ZXing.Net.MAUI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[UIKit.UIView, Microsoft.iOS, Version=16.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065]].SetVirtualView(IView view)
Microsoft.Maui.Handlers.ViewHandler`2[[ZXing.Net.Maui.ICameraBarcodeReaderView, ZXing.Net.MAUI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[UIKit.UIView, Microsoft.iOS, Version=16.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065]].SetVirtualView(IElement view)
Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
Microsoft.Maui.Handlers.ContentViewHandler.UpdateContent(IContentViewHandler handler)
Microsoft.Maui.Handlers.ContentViewHandler.MapContent(IContentViewHandler handler, IContentView page)
Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IContentViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view)
Microsoft.Maui.Handlers.ContentViewHandler.SetVirtualView(IView view)
Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
Microsoft.Maui.Platform.ViewExtensions.ToHandler(IView view, IMauiContext context)
Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.SetPageRenderer(Page page, ShellContent shellContent)
Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.LoadRenderers()
Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.ViewDidLoad()
UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
Program.Main (String[] args)

**Comment by stoff99** [comment]
yes, i have the same. Just turn of the camera permission after first start, restart the app -> boom
I'm currently implementing the permission handling by my own and after the permission is ok, i try to place the scanner by code.

**Comment by jfversluis** [comment]
Closing this as likely addressed by #307, which fixes an iOS disconnect crash when camera permission is denied or the camera session was never fully connected.

Please try the latest build once #307 is available and let us know if you still see an `AVCaptureDeviceInput` disposed-object crash with current code.
